### PR TITLE
[Test] Enable test/profiler/test_profiler.py on XPU

### PR DESCRIPTION
**Pull request by tszulist-hbn**
Part of #142029.

`test/profiler/test_profiler.py` was treated as device-specific as a whole. Most of it is not. Three commits, reviewable in order:

1. Tag all 11 test classes with `hw_classification`, suffix the four CUDA-gated classes that were not already named for it, and move `test_forked_process` out of the accelerator-parameterized class - it was `@onlyOn("cpu")` and never used its `device` argument.
2. Make `setUpModule`'s Kineto priming device-generic. The process-global profiler cannot upgrade from a CPU-only first session, so without priming XPU collection is dead process-wide (`kernel`, `ac2g`, `xpu_runtime` all 0). This commit also carries the `test_forked_process` skip, because generic priming turns that deadlock from order-dependent into unconditional.
3. `allow_xpu=True` on `TestProfilerDevice`, drop a stale skip for the now-closed torch-xpu-ops#2263, widen `test_kineto` to XPU.

### Coverage delta

Intel Data Center GPU Max 1550, oneAPI 2026.1.1.

| XPU test IDs | before | after |
| ------------ | ------ | ----- |
| total        | 0      | 18    |
| passed       | 0      | 16    |
| skipped      | 0      | 2     |

```
$ python -m pytest test/profiler/test_profiler.py -k xpu -v -rs
16 passed, 2 skipped, 143 deselected in 16.33s
```

The full file stays green: `Ran 161 tests ... OK (skipped=60)`.

One coverage loss: `test_forked_process` is skipped on XPU builds. CPU-only and CUDA builds keep it.

### Follow-ups filed against intel/torch-xpu-ops

- intel/torch-xpu-ops#5284 - XPU kernel activities carry `sycl queue` / `l0 queue`
  but no `stream`, `grid` or `block`. Skips `test_kineto_kernel_metadata_in_trace`
  on XPU, and blocks `TestMetadataJsonFormatCUDA` (4 IDs) from being enabled as written.
- intel/torch-xpu-ops#5287 - `os.fork()` deadlocks after Kineto XPU init. Skips
  `test_forked_process` on XPU builds.

Authored with the assistance of an AI coding agent (Claude Code).

### CODE REVIEW

File: test/profiler/test_profiler.py [hunk 1]
```diff
@@ -2834,6 +2796,13 @@ def test_dynamic_toggle(self, device):
     @skipIfTorchDynamo("profiler gets ignored if dynamo activated")
     @onlyOn("cpu")
     @unittest.skipIf(IS_WINDOWS, "can't use os.fork() on Windows")
+    # onlyOn("cpu") means a device_type="xpu" skip would never fire, so this has
```
Comment (pbielak) [inline test/profiler/test_profiler.py]: Is this comment necessary?

If this test was marked as only on CPU, it should be moved to the generic class in my opinion.

The XPU skip is redundant due to `@onlyOn("cpu")`

File: test/profiler/test_profiler.py [hunk 2]
```diff
@@ -3019,6 +2988,11 @@ def test_activity_filter_empty_list(self, device):
         self.assertEqual(len(p.events()), 0)
 
     @skipIfRocm(msg="https://github.com/pytorch/pytorch/issues/180072")
+    @skipIf(
```
Comment (pbielak) [inline test/profiler/test_profiler.py]: Use `skipXPUIf`

File: test/profiler/test_profiler.py
Comment (pbielak) [inline test/profiler/test_profiler.py]: Add CUDA suffix

File: test/profiler/test_profiler.py [hunk 3]
```diff
@@ -3967,6 +3947,8 @@ def handler(prof):
 class TestProfilerEventsParity(TestCase):
```
Comment (pbielak) [inline test/profiler/test_profiler.py]: Add CUDA suffix

File: test/profiler/test_profiler.py
Comment (pbielak) [inline test/profiler/test_profiler.py]: Add CUDA suffix

Comment (pbielak) [inline test/profiler/test_profiler.py]: Add CUDA suffix

**Review by pbielak — CHANGES_REQUESTED** [review]

File: test/profiler/test_profiler.py
Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Done

Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Done

File: test/profiler/test_profiler.py (hunk 3 above)
Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Done

File: test/profiler/test_profiler.py
Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Done

File: test/profiler/test_profiler.py (hunk 2 above)
Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Done

File: test/profiler/test_profiler.py (hunk 1 above)
Comment (tszulist-hbn) [reply test/profiler/test_profiler.py]: Moved to `TestProfiler`, and the `device` arg was never used (the `device_type` it computed was dead), so parameterizing it only bought a dead `_xpu` instance.

The skip stays, though - it isn't redundant. `@onlyOn("cpu")` is what makes the *cpu* instance run, and that's the one that deadlocks: `setUpModule` primes Kineto's XPU profiler for the whole module before any test runs. With the skip
removed and only this test selected it is killed after timeout:

    $ timeout --signal=KILL 90 python -m pytest test/profiler/test_profiler.py -k test_forked_process -v
    Killed

Comment trimmed to two lines now that the `onlyOn` mechanics no longer need explaining.

**Review by pbielak — APPROVED** [review]

**Review by pbielak — APPROVED** [review]

**Comment by kdrozd-dev** [comment]
This looks like a partial duplicate of https://github.com/pytorch/pytorch/pull/193961

File: test/profiler/test_profiler.py [hunk 4]
```diff
@@ -92,18 +94,19 @@ def get_profiler_activities(device_type):
...
-        and ProfilerActivity.CUDA in supported_activities()
-    ):
-        # Kineto's process-global profiler cannot currently upgrade from a
-        # CPU-only first initialization to CUDA-capable profiling. Prime it with
-        # CUDA so CPU-only tests do not poison later CUDA profiler tests.
-        x = torch.ones(1, device="cuda")
-        with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA]):
-            x + x
-            torch.cuda.synchronize()
+    if not kineto_available() or not torch.accelerator.is_available():
```
Comment (etaf) [inline test/profiler/test_profiler.py]: On main this whole block moved to `initialize_kineto_with_cuda()` in torch/testing/_internal/common_profiler.py (#198209). test/profiler/test_cuspy.py and test/test_linalg.py call it too. If you generalize that helper to cover XPU, those two files start priming the XPU profiler at module setup as well, and could pick up the fork-after-Kineto-XPU-init deadlock from intel/torch-xpu-ops#5287 along with it. I haven't checked whether either file forks. It's probably safer to add an accelerator-aware variant and call it only from test_profiler.py, or to check that the other two callers are fine with it.

File: test/profiler/test_profiler.py [hunk 5]
```diff
@@ -3073,6 +3082,10 @@ def test_activity_filter_empty_list(self, device):
         self.assertEqual(len(p.events()), 0)
 
     @skipIfRocm(msg="https://github.com/pytorch/pytorch/issues/180072")
+    @skipXPUIf(
```
Comment (etaf) [inline test/profiler/test_profiler.py]: From the description, this fails on a deterministic assertion (the kernel events have no `stream`/`grid`/`block` args), not a crash or a hang. So it should be an xfail, not a skip. A skip stays silent when intel/torch-xpu-ops#5284 gets fixed. An xfail starts failing then, so someone has to come back and remove it.

```python
@expectedFailureXPU  # https://github.com/intel/torch-xpu-ops/issues/5284
```

`expectedFailureXPU` is in common_device_type.

File: test/profiler/test_profiler.py [hunk 6]
```diff
@@ -3143,11 +3156,13 @@ def test_overhead_activities_own_no_device_time(self, device):
             )
 
 
-instantiate_device_type_tests(TestProfilerDevice, globals())
+instantiate_device_type_tests(TestProfilerDevice, globals(), allow_xpu=True)
```
Comment (etaf) [inline test/profiler/test_profiler.py]: After rebasing, this only enables XPU for TestProfilerDevice. On main, TestExperimentalUtilsDevice, TestProfilerDeviceStopped, TestProfilerEventsParityDevice and TestPythonChromeTraceExport are all ACCELERATOR classes instantiated with `except_for=("cpu",)` and no `allow_xpu`, so XPU is still excluded for them. If the goal is enabling the file on XPU, those need `allow_xpu=True` too, or the description should say they're out of scope.

**Review by etaf — CHANGES_REQUESTED** [review]
